logstash-core/lib/logstash/config/source/local.rb

```diff
@@ -32,23 +32,29 @@ module LogStash module Config module Source
   #
   class Local < Base
     class ConfigStringLoader
+      include LogStash::Util::SubstitutionVariables
+      include LogStash::Util::Loggable
+
       INPUT_BLOCK_RE = /input *{/
       OUTPUT_BLOCK_RE = /output *{/
       EMPTY_RE = /^\s*$/
 
-      def self.read(config_string)
-        config_parts = [org.logstash.common.SourceWithMetadata.new("string", "config_string", 0, 0, config_string)]
+      def initialize(config_string)
+        @config_string = config_string
+      end
+
+      def read
+        config_parts = [org.logstash.common.SourceWithMetadata.new("string", "config_string", 0, 0, deep_replace(@config_string))]
```

Comment:
since we are always dealing with a `String`, `SubstitutionVariables#replace_placeholders` may be a better fit than `SubstitutionVariables#deep_replace`.